### DISCUSSION TITLE
Troubleshooting geo location attendance on android phones.

### DISCUSSION THREAD

**Discussion by yogeshkudikala** [madrecha/jamku#1123] [QA] [answered]
Sometime mobile app does not capture correct location of the user while marking attendance. How to troubleshoot this issue?

**Comment by yogeshkudikala** [comment, marked as answer]
When you visit the attendance location page for the first time it will ask you for location permission, do enable it. Make sure that the page does not display any warnings to you.
![Screenshot_2022-09-10-17-45-09-58_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/51023000/190954288-0c473a75-f143-4d6f-9d1a-b157fb1bc70e.jpeg)
![Screenshot_2022-09-10-17-45-04-71_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/51023000/190954299-ea5cfb34-c9ba-49d5-a104-3544f390541f.jpeg)

Check if your device location is ON. If it is not then switch on the device location and refresh the attendance location page by clicking on the refresh icon present in the breadcrumbs.
![Screenshot_2022-09-10-17-43-32-17_b783bf344239542886fee7b48fa4b892](https://user-images.githubusercontent.com/51023000/190954477-1791edf4-5cdb-4518-aa52-fd423d958bd6.jpeg)

If the page still does not show the correct location, follow the below steps.
Step 1: Visit Chrome browser 
![image](https://user-images.githubusercontent.com/51023000/190954519-038dcffe-94fa-46c9-ba78-905efd9eae62.png)

Step 2: Open Chrome Settings page
![Screenshot_2022-09-10-17-43-45-75_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/51023000/190954943-ccdabb8e-7f18-4896-be59-7597718c9bf6.jpeg)
![Screenshot_2022-09-10-17-43-50-12_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/51023000/190954972-eefae968-b311-451d-a760-6c7e173718d9.jpeg)

Step 3: Click on "Site Settings" option
![Screenshot_2022-09-10-17-43-57-65_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/51023000/190955178-c21866b6-4db8-4552-976d-00c0a6482d15.jpeg)

Step 4: Click on "Location". Make sure that "Jamku" has been given location permission. If "Jamku" is under block section, kindly unblock it.
![Screenshot_2022-09-10-17-44-05-83_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/51023000/190955188-b37aedfe-380d-49d3-8d3f-c659bf4f14ef.jpeg)
![image](https://user-images.githubusercontent.com/51023000/190955230-dc07b1fe-e8f8-44d4-a519-cba29d8a882a.png)

Once done. Refresh the attendance location page.

**Comment by adarshmadrecha** [comment]
Basically there are 3 prerequisites for marking Geo Location attendance

1. On Mobile, location should be ON
2. Location permission is allowed on Chrome/Default browser.
3. jamku.app `site` should be location access.

## 1
<img src="https://user-images.githubusercontent.com/11911938/212036753-bb5c053b-0d02-4218-a66c-58f46dcd0098.jpg" width="300px"> 

## 2
<img src="https://user-images.githubusercontent.com/11911938/212036798-65b47227-0b29-40d7-8ed2-84cf867884d1.jpg" width="300px"> 

## 3
<img src="https://user-images.githubusercontent.com/11911938/212036883-b3f5e570-0f48-4936-99d7-0b7c87e18c4b.jpg" width="300px">
